rebasehelper/specfile.py

```diff
@@ -102,6 +103,85 @@ def get_strip(self):
         return self.strip
 
 
+class SpecContent(object):
+
+    """Class representing content of a SPEC file."""
+
+    defined_section_headers = [
+        '%package',
+        '%prep',
+        '%build',
+        '%install',
+        '%check',
+        '%clean',
+        '%prerun',
+        '%postrun',
+        '%pretrans',
+        '%posttrans',
+        '%pre',
+        '%post',
+        '%files',
+        '%changelog',
+        '%description',
+        '%triggerpostun',
+        '%triggerprein',
+        '%triggerun',
+        '%triggerin',
+        '%trigger',
+        '%verifyscript',
+        '%sepolicy',
+        '%filetriggerin',
+        '%filetrigger',
+        '%filetriggerun',
+        '%filetriggerpostun',
+        '%transfiletriggerin',
+        '%transfiletrigger',
+        '%transfiletriggerun',
+        '%transfiletriggerpostun',
+    ]
+
+    def __init__(self, content):
+        self.sections = CaseInsensitiveDict()
+        self._split_sections(content)
+
+    def __str__(self):
+        """Join SPEC file sections back together."""
+        content = []
+        for header, section in six.iteritems(self.sections):
+            if header != '%package':
+                content.append(header + '\n')
+            for line in section:
+                content.append(line + '\n')
+        return ''.join(content)
+
+    def _split_sections(self, content):
+        """Splits content of a SPEC file into sections.
+
+        Args:
+            content (str): Content of the SPEC file
+
+        """
+        lines = content.splitlines()
+        section_headers_re = [re.compile(r'^{0}.*'.format(x), re.I) for x in self.defined_section_headers]
```

Comment:
You should always use `re.escape()` when constructing regexp from variables. Though it's not strictly necessary here, it's a good practice to follow.